AddonBicepTemplate/FullAzureMonitorMetricsProfile.bicep

```diff
@@ -270,6 +280,14 @@ resource kubernetesRecordingRuleGroup 'Microsoft.AlertsManagement/prometheusRule
   }
 }
 
+resource roleNameGuid_resource 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
+  name: roleNameGuid
+  properties: {
```

Comment:
Currently in bicep, there is no way to explicitly "scope" the role assignment on a "resource id"(like in Arm template). Bicep expects a value of type "resource | tenant" and currently there is no resource spec for Azure Monitor Workspace. So, as a workaround, the default scoping is on the resource group and thus the monitor reader role is applied on the same Azure monitor workspace(by inheritence) which is the expected behavior. I have tested it in my workspace [here](https://ms.portal.azure.com/#@microsoft.onmicrosoft.com/resource/subscriptions/ce4d1293-71c0-4c72-bc55-133553ee9e50/resourceGroups/sohamTestKV/providers/microsoft.monitor/accounts/sohambiceptemplate/users).

Reply:
Sounds good, we can mention this as a limiation in the public preview doc update that you make.

Reply:
Yes. I will mention this. Thanks!